### PR TITLE
Port changes to adapt to Cdnjs API change

### PR DESCRIPTION
**Pull request by jimmylewis**
Summary of the changes:
 - Switches away from relying on "assets" including information for all library versions.
 - Add user agent to the web requests.

### CODE REVIEW

File: src/LibraryManager/Providers/Cdnjs/CdnjsCatalog.cs [hunk 1]
```diff
@@ -142,20 +143,22 @@ public async Task<ILibrary> GetLibraryAsync(string libraryId, CancellationToken
...
+            if (!(await GetLibraryVersionsAsync(name, cancellationToken)).Contains(version))
+            {
+                throw new InvalidLibraryException(libraryId, _provider.Id);
+            }
+
             try
             {
-                IEnumerable<Asset> assets = await GetAssetsAsync(name, cancellationToken).ConfigureAwait(false);
-                Asset asset = assets.FirstOrDefault(a => a.Version == version);
+                JObject groupMetadata = await GetLibraryGroupMetadataAsync(name, cancellationToken);
```
Comment (ToddGrun) [inline src/LibraryManager/Providers/Cdnjs/CdnjsCatalog.cs:153]: >; [](http://example.com/codeflow?start=99&length=1)

ConfigureAwait?

File: src/LibraryManager/Providers/Cdnjs/CdnjsCatalog.cs [hunk 2]
```diff
@@ -269,54 +272,70 @@ private async Task<bool> EnsureCatalogAsync(CancellationToken cancellationToken)
...
+            return null;
         }
 
-        private async Task<IEnumerable<Asset>> GetAssetsAsync(string groupName, CancellationToken cancellationToken)
+        private async Task<IEnumerable<string>> GetLibraryFilesAsync(string libraryName, string version, CancellationToken cancellationToken)
         {
-            var list = new List<Asset>();
-            string localFile = Path.Combine(_provider.CacheFolder, groupName, "metadata.json");
-            string url = string.Format(_metaPackageUrlFormat, groupName);
+            string localFile = Path.Combine(_provider.CacheFolder, libraryName, $"{version}-metadata.json");
```
Comment (ToddGrun) [inline src/LibraryManager/Providers/Cdnjs/CdnjsCatalog.cs:307]: >son"); [](http://example.com/codeflow?start=102&length=6)

nit: extra newline

**Review by ToddGrun — DISMISSED** [review]
:shipit:

File: src/LibraryManager/Providers/Cdnjs/CdnjsCatalog.cs (hunk 2 above)
Comment (jimmylewis) [reply src/LibraryManager/Providers/Cdnjs/CdnjsCatalog.cs:307]: Not seeing it, is this a CodeFlow artifact?

Comment (ToddGrun) [reply src/LibraryManager/Providers/Cdnjs/CdnjsCatalog.cs:307]: ok, this isn't where I marked it, I still see it in the try block though

Comment (jimmylewis) [reply src/LibraryManager/Providers/Cdnjs/CdnjsCatalog.cs:307]: Ah, got it

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**